### PR TITLE
Added email_notif option (#80)

### PR DESCRIPTION
**Pull request by lily-wittle**

### CODE REVIEW

**Comment by lily-wittle** [comment]
#80

File: create_batch.py [hunk 1]
```diff
@@ -57,11 +57,12 @@ def parse_arguments(args):
     return job_submission_params
 
 
-def get_job_file_lines(job_name, job_file_name, scheduler=None, memory=3600, walltime="4:00", queue=None):
+def get_job_file_lines(job_name, job_file_name, email_notif, scheduler=None, memory=3600, walltime="4:00", queue=None):
```
Comment (gravelcycles) [inline create_batch.py:60]: What's the reason to default `email_notif` to `False` in some cases but not all? My thought is the desired functionality would be to always default to `False`.

**Comment by falkamelung** [comment]
Just a quick thought - ignore if it is nonsense.

I have not looked how it is implemented, but would it make sense to set email_notif when submit_script is called?


cb.submit_script(inps.project_name, job_file_name, email_notif = True, sys.argv[:], inps.work_dir, wall_time)


------------------------------------------------------
Falk Amelung
Professor
Department of Marine Geosciences
Rosenstiel School of Marine and Atmospheric Sciences
University of Miami
4600 Rickenbacker Causeway
Miami, FL 33149
USA
Tel: 305 421 4949
E-mail: famelung@rsmas.miami.edu<mailto:famelung@rsmas.miami.edu>
Web: http://insar.rsmas.miami.edu
InSAR data:
http://insarmaps.miami.edu




On Mar 10, 2019, at 10:54 AM, David W Grossman <notifications@github.com<mailto:notifications@github.com>> wrote:


@2gotgrossman commented on this pull request.

________________________________

In create_batch.py<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fgeodesymiami%2Frsmas_isce%2Fpull%2F92%23discussion_r264042762&data=02%7C01%7Cfamelung%40rsmas.miami.edu%7C2c189756b7094cbcb55008d6a5683ba2%7C2a144b72f23942d48c0e6f0f17c48e33%7C0%7C0%7C636878264439312378&sdata=9eSzxBNnSxwscNPlIgh3%2BkFLBLIe%2BgOc2ooUtyHAZiw%3D&reserved=0>:

> @@ -57,11 +57,12 @@ def parse_arguments(args):
     return job_submission_params


-def get_job_file_lines(job_name, job_file_name, scheduler=None, memory=3600, walltime="4:00", queue=None):
+def get_job_file_lines(job_name, job_file_name, email_notif, scheduler=None, memory=3600, walltime="4:00", queue=None):


What's the reason to default email_notif to False in some cases but not all? My thought is the desired functionality would be to always default to False.

File: create_batch.py (hunk 1 above)
Comment (lily-wittle) [reply create_batch.py:60]: @2gotgrossman I wrote it to default to `False` when submitting a batch of jobs, but to `True` when submitting major scripts based on what Falk had said in the issue ("For create_batch.py the default should be WITHOUT email notification, for process_rsmas the default should be WITH email notification"). But that could be wrong if it is supposed to only be `True` for `process_rsmas.py` and `False` for other major scripts

**Comment by lily-wittle** [comment]
@falkamelung with the way it's written right now you could set `email_notif` when `submit_script` is called (`submit_script(job_name, job_file_name, argv, work_dir, walltime, email_notif)`) if you want to override the default, which is `True` 

related to both this and David's comment: based on what you said in the issue ("For create_batch.py the default should be WITHOUT email notification, for process_rsmas the default should be WITH email notification") I wrote the default to be `False` when submitting batch jobs and `True` when submitting a major script. Is this what you want? Or should it default to `False` for `submit_script` -- is `process_rsmas.py` the only script for which you want email notifications?

**Comment by falkamelung** [comment]
Yes, exactly. Thank you. Only I have the feeling that sometimes the emailing does not work but this is probably a pegasus problem.
------------------------------------------------------
Falk Amelung
Professor
Department of Marine Geosciences
Rosenstiel School of Marine and Atmospheric Sciences
University of Miami
4600 Rickenbacker Causeway
Miami, FL 33149
USA
Tel: 305 421 4949
E-mail: famelung@rsmas.miami.edu<mailto:famelung@rsmas.miami.edu>
Web: http://insar.rsmas.miami.edu
InSAR data:
http://insarmaps.miami.edu




On Mar 18, 2019, at 7:00 PM, Lily Wittle <notifications@github.com<mailto:notifications@github.com>> wrote:


@falkamelung<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ffalkamelung&data=02%7C01%7Cfamelung%40rsmas.miami.edu%7C4f2b9c64d5d342b93efb08d6abfddfeb%7C2a144b72f23942d48c0e6f0f17c48e33%7C0%7C0%7C636885504201930014&sdata=gKdU6UT8lcgSvW3E%2B7UIZeNTV7h9QGVl28OiicKU7KA%3D&reserved=0> with the way it's written right now you could set email_notif when submit_script is called (submit_script(job_name, job_file_name, argv, work_dir, walltime, email_notif)) if you want to override the default, which is True

related to both this and David's comment: based on what you said in the issue ("For create_batch.py the default should be WITHOUT email notification, for process_rsmas the default should be WITH email notification") I wrote the default to be False when submitting batch jobs and True when submitting a major script. Is this what you want? Or should it default to False for submit_script -- is process_rsmas.py the only script for which you want email notifications?

**Pull request merged**